internal/rbd/rbd_util.go

```diff
@@ -229,6 +230,13 @@ func createImage(ctx context.Context, pOpts *rbdVolume, cr *util.Credentials) er
 		return fmt.Errorf("failed to create rbd image: %w", err)
 	}
 
+	if pOpts.ThickProvision {
+		err = pOpts.allocate()
```

Comment:
if the allocation fails here we are not cleaning up the volume but we are returning the error, as the error is returned the entry in the OMAP will be deleted, and then this can leave stale rbd images.

Reply:
good catch, will look into that too